### PR TITLE
pet: Add support for tags in snippets

### PR DESCRIPTION
**Pull request by tams**
### Description

Add missing `tag` option for elements of `programs.pet.snippets`.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added.

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

### CODE REVIEW

File: tests/modules/programs/pet/snippets.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        command = "git log -p -G <regex>";
+        tag = [ "git" "regex" ];
+      }];
+    };
+
+    nmt.script = ''
+      assertFileContent home-files/.config/pet/snippet.toml ${./snippet.toml}
+    '';
+  };
+}
```
Comment (berbiche) [inline tests/modules/programs/pet/snippets.nix:20]: We don't need to fetch packages in tests so we make dummy, empty packages:

```suggestion
{
  config = {
    programs.pet = {
      enable = true;
      selectcmdPackage = pkgs.writeScriptBin "pet-cmd" "" // { outPath = "@pet-cmd@"; };
      snippets = [{
        description = "git: search full history for regex";
        command = "git log -p -G <regex>";
        tag = [ "git" "regex" ];
      }];
    };
    
    nixpkgs.overlays = [
      (self: super: { pet = pkgs.writeScriptBin "pet" "" // { outPath = "@pet@"; }; })
    ];

    nmt.script = ''
      assertFileContent home-files/.config/pet/snippet.toml ${./snippet.toml}
    '';
  };
}
```

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Could you please add the suggested changes to the test case and run the formatter again?

File: tests/modules/programs/pet/snippets.nix (hunk 1 above)
Comment (tams) [reply tests/modules/programs/pet/snippets.nix:20]: Duly noted. 🎉

**Review by berbiche — APPROVED** [review]
LGTM!

File: modules/programs/pet.nix [hunk 2]
```diff
@@ -36,6 +36,15 @@ let
           Example output of the command.
         '';
       };
+
+      tag = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = ''["git" "nixpkgs"]'';
```
Comment (berbiche) [inline modules/programs/pet.nix:43]:
```suggestion
        example = literalExample ''["git" "nixpkgs"]'';
```

Comment (berbiche) [reply modules/programs/pet.nix:43]: I missed this

**Review by berbiche — APPROVED** [review]
